Modules/Core/Mesh/include/itkMesh.hxx

```diff
@@ -103,6 +103,160 @@ Mesh<TPixelType, VDimension, TMeshTraits>::SetCells(CellsContainer * cells)
+  {
+    auto         cellPointer = cellItr->Value();
+    unsigned int numOfPoints = cellPointer->GetNumberOfPoints();
+
+    // Insert the cell type
+    cellOutputVectorContainer->InsertElement(index++, static_cast<unsigned long int>(cellPointer->GetType()));
+    // Insert the number of points in the cell
+    cellOutputVectorContainer->InsertElement(index++, numOfPoints);
+
+    auto pointIds = cellPointer->GetPointIds();
+
+    // Insert the points in the cell
+    for (unsigned int i = 0; i < numOfPoints; ++i)
+    {
+      cellOutputVectorContainer->InsertElement(index++, pointIds[i]);
+    }
+  }
+
+  return cellOutputVectorContainer;
+}
+
+// Helper function to create a new cell of a given type
+template <typename TPixelType, unsigned int VDimension, typename TMeshTraits>
+void
+Mesh<TPixelType, VDimension, TMeshTraits>::CreateCell(int cellType, CellAutoPointer & cellPointer)
+{
+  auto cellTypeEnum = static_cast<CellGeometryEnum>(cellType);
+
+  switch (cellTypeEnum)
+  {
+    case CellGeometryEnum::VERTEX_CELL:
+      cellPointer.TakeOwnership(new OutputVertexCellType);
+      break;
+    case CellGeometryEnum::LINE_CELL:
+      cellPointer.TakeOwnership(new OutputLineCellType);
+      break;
+    case CellGeometryEnum::TRIANGLE_CELL:
+      cellPointer.TakeOwnership(new OutputTriangleCellType);
+      break;
+    case CellGeometryEnum::QUADRILATERAL_CELL:
+      cellPointer.TakeOwnership(new OutputQuadrilateralCellType);
+      break;
+    case CellGeometryEnum::POLYGON_CELL:
+      cellPointer.TakeOwnership(new OutputPolygonCellType);
+      break;
+    case CellGeometryEnum::TETRAHEDRON_CELL:
+      cellPointer.TakeOwnership(new OutputTetrahedronCellType);
+      break;
+    case CellGeometryEnum::HEXAHEDRON_CELL:
+      cellPointer.TakeOwnership(new OutputHexahedronCellType);
+      break;
+    case CellGeometryEnum::QUADRATIC_EDGE_CELL:
+      cellPointer.TakeOwnership(new OutputQuadraticEdgeCellType);
+      break;
+    case CellGeometryEnum::QUADRATIC_TRIANGLE_CELL:
+      cellPointer.TakeOwnership(new OutputQuadraticTriangleCellType);
+      break;
+    default:
+      itkExceptionMacro(<< "Unknown mesh cell");
+  }
+
+  return;
+}
+
+/**
+ * Convenience method to set the cells container using 1d vector.
+ * First element of the vector is the cell type and next is number of
+ * points in the cell followed by the point ids in that cell.
+ * Can cause exception if input cell array isn't of right dimension.
+ */
+template <typename TPixelType, unsigned int VDimension, typename TMeshTraits>
+void
+Mesh<TPixelType, VDimension, TMeshTraits>::SetCellsUsingArray(CellsVectorContainer * cells)
+{
+  itkDebugMacro("setting Cells container to " << cells);
+
+  this->ReleaseCellsMemory();
+  unsigned long index = 0;
+  unsigned long cellId = 0;
+
+  while (index < cells->Size())
+  {
+    int currentCellType = static_cast<int>(cells->GetElement(index++));
```

Comment:
Should we always cast to `int`? Not `CellIdentifier` or something else?

Reply:
That element will always be a cell type which are limited in number (~ 10) so int is fine there.

The point ids in that cell are taken in this line 
`cellPointer->SetPointId(i, cells->GetElement(index++));`

And the cell is inserted at:
`this->m_CellsContainer->InsertElement(cellId++, cellPointer.ReleaseOwnership());`
where the cellId is unsigned long. I can replace it to CellIdentifier.

Reply:
Used IdentifierType where appropriate.